controller/TaskQueueWebApi.php

```diff
@@ -73,15 +77,26 @@ public function getAll()
             $offset = (int) $this->getRequestParameter(self::PARAMETER_OFFSET);
         }
 
-        if ($this->hasRequestParameter(self::PARAMETER_REPORT_INCLUDED)) {
-            $reportIncluded = (bool) $this->getRequestParameter(self::PARAMETER_REPORT_INCLUDED);
-        }
+        /** @var FileSystemService $fs */
+        $fs = $this->getServiceManager()->get(FileSystemService::SERVICE_ID);
+
+        // adding hasFile to the results
+        $collection = new HasFileCollectionDecorator(
```

Comment:
I think this collection approach can have performance issues. (for each collection you iterate and only add one value, as more collection iterators we have as slow this can became).
If you remember we discuss last PR to have a general CollectionDecorator who receives entities decorators and just apply the decorate for each entity.
You can achieve  this easy:
Have a CollectionDecorator which will be constructed with $data, and array (collection) of entity decorators. when method toArray() called the array data will be transformed to entities decorated with the decorators.
This means that the search method will return TaskLogCollectionInterface which it's extended by CollectionDecorator.
So basically the search will return as well a collection in your case will be CollectionDecorator. 
If you pass no decorators to the CollectionDecorator the outcome will be the same (default behavior).